ui/page/dexclient/markets.go

```diff
@@ -79,13 +101,50 @@ func (pg *Page) Layout(gtx C) D {
 		}
 	default:
 		body = func(gtx C) D {
-			return pg.pageSections(gtx, pg.registrationStatusLayout)
+			dex := pg.dex()
+			if !dex.Connected {
+				return D{}
+			}
+
+			if dex.PendingFee == nil && pg.miniTradeFormWdg != nil {
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						return pg.pageSections(gtx, pg.miniTradeFormWdg.layout)
+					}),
+					layout.Rigid(pg.settingsLayout),
+				)
+			}
+
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return pg.pageSections(gtx, pg.registrationStatusLayout)
+				}),
+				layout.Rigid(pg.settingsLayout),
+			)
 		}
 	}
 
 	return components.UniformPadding(gtx, body)
 }
 
+func (pg *Page) settingsLayout(gtx C) D {
```

Comment:
The `registrationStatusLayout` is referred to first (line 120) before the `settingsLayout`. Re-order the methods to position the `registrationStatusLayout` before the `settingsLayout`. It's easier to follow the code when methods are defined close to where they are called.